enhancements/sig-architecture/33-detached-deploy-mode/detached-deploy-mode.md

```diff
@@ -0,0 +1,224 @@
+# Running deployments of cluster-manager and klusterlet outside of Hub and ManagedCluster
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [website](https://github.com/open-cluster-management/website/)
+
+## Summary
+
+The proposed work would support running deployments of cluster-manager and klusterlet outside of Hub and ManagedCluster. 
+
+## Motivation
+
+Currently, the cluster manager and klusterlet should be deployed on the hub/managed cluster, it is mandatory to expose the hub cluster to the managed cluster even if the managed cluster resides in an insecure environment. 
+
+To address this, we propose to provide an option for running cluster-manager/klusterlet outside of the hub/managed cluster.
+
+This ability will bring some additional possible benefits:
+* We can leverage compute resources better by running several cluster-managers/klusterlets deployments on the same powerful cluster; it also enhances our management of all deployments.
+* The footprints of the hub/managed cluster can be reduced.
+* We can use this ability to run the klusterlet on the hub cluster, so we do not need to expose the hub cluster to the managed cluster, and accordingly, we expose the managed cluster to the hub (A bit like Pull Mode to Push Mode).
```

Comment:
I wonder if this approach makes the management cluster a performance bottleneck? What is the advantage of running Klusterlets in a management cluster vs a pure push model?

Reply:
We had a case that a customer has a super-powerful machine and want to use it to run every deployment of each hub cluster. So base on this situation, the management cluster would be a bottleneck, but if the situation changes, we might need to use another approach. 
For the advantage, I would say the compatibility since there are no API changes. @zhujian7  Feel free to add more.
And CSR is still needed for this.

Reply:
@xuezhaojun I think it worth writing a section to compare it with push mode (cons and pros). I think a push mode will need listwatch on multiple spokes, which has been proven to be not scalable in kubefed since a single controller will consume a lot of memory. Running klusterlet separated distribute the computation to multiple pods. It might be good to consider even in push case, we still have workers for each different spoke.

Reply:
The content added.

Reply:
good point, the push model shouldn't require to run all of them in the same pod.